llvm/lib/Target/X86/X86TargetTransformInfo.cpp

```diff
@@ -1564,6 +1564,7 @@ InstructionCost X86TTIImpl::getShuffleCost(
   bool IsInLaneShuffle = false;
   if (BaseTp->getPrimitiveSizeInBits() > 0 &&
       (BaseTp->getPrimitiveSizeInBits() % 128) == 0 &&
+      BaseTp->getScalarSizeInBits() == LT.second.getScalarSizeInBits() &&
```

Comment:
Will it work for pointers?

Reply:
Not, thats what `BaseTp->getPrimitiveSizeInBits() > 0` is checking for as ptr vectors return 0